src/pixelator/cli/collapse.py

```diff
@@ -25,6 +26,38 @@
 )
 
 
+def _handle_errors(jobs, executor):
+    for job in jobs:
+        exception = job.exception()
+        if exception is None:
+            continue
+
+        logger.error(
+            "Found an issue in the process pool. Trying to determine what went wrong and set the correct exit code. Exception was: %s",
+            exception,
+        )
+        process_map = executor._processes
+        for pid in process_map.keys():
+            exit_code = process_map[pid].exitcode
+            if exit_code is not None and exit_code != 0:
+                logger.error(
+                    "The child process in the process pool returned a non-zero exit code: %s.",
+                    exit_code,
+                )
+                # If we have an out of memory exception, make sure we exit with that.
+                if abs(exit_code) == 9:
+                    logger.error(
+                        "One of the child processes was killed (exit code: 9). Usually this is caused "
+                        "by a child process using to much memory. We will return an exit code of 137 "
+                        "to indicate that the process was killed by the out of memory killer."
+                    )
```

Comment:
I think this is currently slightly inaccurate since anyone can send kill signals.
Maybe something like this?

```suggestion
                        "One of the child processes was killed (exit code: 9). "
                        "Usually this is caused by the out-of-memory killer terminating the process. "
                        "The parent process will return an exit code of 137 to indicate that it terminated because of a kill signal in the child process."
                    )
```